From 5cb04d9f3b7b05ae2579acca6db2063ab38b32d1 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Fri, 3 Jul 2020 10:32:41 +0800
Subject: [PATCH] policy/modules/services/radius: fix avc denials for
 freeradius

* Add getcap, execmem and ptrace to radiusd_t process
* Add dac_read_search capability to radiusd_t
* Allow radiusd search tmpfs dirs

Fixes:
avc:  denied  { getcap } for  pid=299 comm="radiusd"
scontext=system_u:system_r:radiusd_t:s0-s15:c0.c1023
tcontext=system_u:system_r:radiusd_t:s0-s15:c0.c1023 tclass=process
permissive=0

avc:  denied  { execmem } for  pid=301 comm="radiusd"
scontext=system_u:system_r:radiusd_t:s0-s15:c0.c1023
tcontext=system_u:system_r:radiusd_t:s0-s15:c0.c1023 tclass=process
permissive=0

avc:  denied  { ptrace } for  pid=445 comm="radiusd"
scontext=system_u:system_r:radiusd_t:s0-s15:c0.c1023
tcontext=system_u:system_r:radiusd_t:s0-s15:c0.c1023 tclass=process
permissive=0

avc:  denied  { dac_read_search } for  pid=299 comm="radiusd"
capability=2  scontext=system_u:system_r:radiusd_t:s0-s15:c0.c1023
tcontext=system_u:system_r:radiusd_t:s0-s15:c0.c1023 tclass=capability
permissive=0

avc:  denied  { search } for  pid=299 comm="radiusd" name="/"
dev="tmpfs" ino=12394
scontext=system_u:system_r:radiusd_t:s0-s15:c0.c1023
tcontext=system_u:object_r:tmpfs_t:s0 tclass=dir permissive=0

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/radius.te | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/policy/modules/services/radius.te b/policy/modules/services/radius.te
index 56cb430a5..2e7e336c9 100644
--- a/policy/modules/services/radius.te
+++ b/policy/modules/services/radius.te
@@ -32,9 +32,9 @@ files_type(radiusd_var_lib_t)
 # Local policy
 #
 
-allow radiusd_t self:capability { chown dac_override fsetid kill setgid setuid sys_resource sys_tty_config };
+allow radiusd_t self:capability { chown dac_override dac_read_search fsetid kill setgid setuid sys_resource sys_tty_config };
 dontaudit radiusd_t self:capability sys_tty_config;
-allow radiusd_t self:process { getsched setrlimit setsched sigkill signal };
+allow radiusd_t self:process { getcap getsched setrlimit setsched sigkill signal execmem ptrace };
 allow radiusd_t self:fifo_file rw_fifo_file_perms;
 allow radiusd_t self:unix_stream_socket { accept listen };
 allow radiusd_t self:tcp_socket { accept listen };
@@ -93,6 +93,7 @@ domain_use_interactive_fds(radiusd_t)
 
 fs_getattr_all_fs(radiusd_t)
 fs_search_auto_mountpoints(radiusd_t)
+fs_search_tmpfs(radiusd_t)
 
 files_read_usr_files(radiusd_t)
 files_read_etc_runtime_files(radiusd_t)
-- 
2.17.1

